luigi/execution_summary.py

```diff
@@ -30,36 +30,93 @@
 
 class execution_summary(luigi.Config):
     summary_length = luigi.IntParameter(default=5)
-    legacy_run_result = luigi.BoolParameter(default=True)
+
+
+class LuigiRetCodes:
+    """
+    All possible return codes for **LuigiRunResult.status** when the argument ``detailed_summary=True`` in
+    *luigi.run() / luigi.build*. Here are the codes and what they mean:
+
+    =============================  =====  ========================================================== 
+    Return Code Name               Value  Meaning 
+    =============================  =====  ========================================================== 
+    SUCCESS                        0      There were no failed tasks or missing dependencies 
+    SUCCESS_WITH_RETRY             1      There were failed tasks but they all succeeded in a retry 
+    FAILED                         2      There were failed tasks
+    FAILED_AND_SCHEDULING_FAILED   3      There were failed tasks and tasks whose scheduling failed
+    SCHEDULING_FAILED              4      There were tasks whose scheduling failed
+    NOT_RUN                        5      There were tasks that were not granted run permission by the scheduler
+    MISSING_EXT                    6      There were missing external dependencies
+    =============================  =====  ==========================================================
+    
+    """
+    SUCCESS = 0
+    SUCCESS_WITH_RETRY = 1
+    FAILED = 2
+    FAILED_AND_SCHEDULING_FAILED = 3
+    SCHEDULING_FAILED = 4
+    NOT_RUN = 5
+    MISSING_EXT = 6
+
+    # Make sure all the retcodes have a (smiley, reason) in this dictionary
+    _DETAILS = {
+        SUCCESS: (":)", "there were no failed tasks or missing dependencies"),
+        SUCCESS_WITH_RETRY: (":)", "there were failed tasks but they all succeeded in a retry"),
+        FAILED: (":(" ,"there were failed tasks"),
+        FAILED_AND_SCHEDULING_FAILED: (":(", "there were failed tasks and tasks whose scheduling failed"),
+        SCHEDULING_FAILED: (":(", "there were tasks whose scheduling failed"),
+        NOT_RUN: (":|", "there were tasks that were not granted run permission by the scheduler"),
+        MISSING_EXT: (":|", "there were missing external dependencies")
+    }
```

Comment:
Can you use a real enum and then use this kind of syntax? Would it be better?

https://docs.python.org/3/library/enum.html#planet

Reply:
Using `Enum` as the base class would have some advantages like making the enums iterable and uniqueness-validation etc. I'll use it as the base class 👍 
.
.
@Tarrasch Since we're exposing the status (enum) as a part of `LuigiRunResult`, I was thinking of returning an integer value for the enum instead of a tuple.  So I kept the enums as integers themselves and  kept this dictionary of descriptions for internal logging etc. Two questions:
- Should I make the enums return a tuple instead of the integer then ?
- If you noticed, I renamed this enum class to `LuigiStatusCode` to avoid confusion with the existing `retcodes`. Should I rename it back to `LuigiRetCode` ? ☕️

Reply:
> - Should I make the enums return a tuple instead of the integer then ?

I made the enums tuples 👍 So I assumed, people would need to access the code outside the library, so I added `LuigiStatusCode ` to luigi's `__init__` as well. 

> - If you noticed, I renamed this enum class to LuigiStatusCode ..

Awaiting your response on this one ☝️